# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalChild.h
# Commit: 40b61344f0da
# Full Hash: 40b61344f0da14f9700a75a8a1c67458e440b68a
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1f - Implement BrowsingContext::LoadURI. r=nika
#   
#   In order to do cross-process targeted window.open() and link click operations,
#   we need a way to load URIs in the current DocShell of a BrowsingContext,
#   whichever process it lives in.
# ==============================================================================

diff -r 3bffe760e2db -r 40b61344f0da dom/ipc/WindowGlobalChild.h
--- a/dom/ipc/WindowGlobalChild.h	Thu Aug 01 16:36:32 2019 -0700
+++ b/dom/ipc/WindowGlobalChild.h	Thu Aug 01 18:00:32 2019 -0700
@@ -114,6 +114,8 @@
   mozilla::ipc::IPCResult RecvRawMessage(const JSWindowActorMessageMeta& aMeta,
                                          const ClonedMessageData& aData);
 
+  mozilla::ipc::IPCResult RecvLoadURIInChild(nsDocShellLoadState* aLoadState);
+
   mozilla::ipc::IPCResult RecvChangeFrameRemoteness(
       dom::BrowsingContext* aBc, const nsString& aRemoteType,
       uint64_t aPendingSwitchId, ChangeFrameRemotenessResolver&& aResolver);